// @progress/kendo-angular-charts/dist/es/common/settings.component.js
import { Change } from './configuration.service';
import { copyChanges } from './copy-changes';
/**
 * @hidden
 */
var SettingsComponent = /** @class */ (function () {
    function SettingsComponent(configKey, configurationService) {
        this.configKey = configKey;
        this.configurationService = configurationService;
        this.store = {};
        if (configKey === undefined) {
            throw new Error('Configuration key not set');
        }
    }
    SettingsComponent.prototype.ngOnDestroy = function () {
        this.store = undefined;
        this.notify();
    };
    SettingsComponent.prototype.ngOnChanges = function (changes) {
        copyChanges(changes, this.store);
        this.notify();
    };
    SettingsComponent.prototype.markAsVisible = function () {
        this.store.visible = true;
        this.notify();
    };
    SettingsComponent.prototype.notify = function () {
        this.configurationService.notify(new Change(this.configKey, this.store));
    };
    return SettingsComponent;
}());
export { SettingsComponent };
