// @progress/kendo-angular-charts/dist/npm/common/chart-default-theme.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var kendo_charts_1 = require("@progress/kendo-charts");
/** @hidden */
exports.chartDefaultTheme = function () {
    return Object.assign({}, kendo_charts_1.chartBaseTheme(), {
        axisDefaults: {
            crosshair: {
                color: 'rgba(0, 0, 0, 0.5)'
            },
            labels: {
                color: 'rgb(101, 101, 101)',
                font: '12px serif'
            },
            line: {
                color: 'rgba(0, 0, 0, 0.08)'
            },
            majorGridLines: {
                color: 'rgba(0, 0, 0, 0.08)'
            },
            minorGridLines: {
                color: 'rgba(0, 0, 0, 0.04)'
            },
            notes: {
                icon: {
                    background: 'rgba(0, 0, 0, 0.5)',
                    border: {
                        color: 'rgba(0, 0, 0, 0.5)'
                    }
                },
                line: {
                    color: 'rgba(0, 0, 0, 0.5)'
                },
                label: {
                    font: '14px serif'
                }
            },
            title: {
                color: 'rgb(101, 101, 101)',
                font: '14px serif'
            }
        },
        chartArea: {
            background: 'rgb(255, 255, 255)'
        },
        legend: {
            inactiveItems: {
                labels: {
                    color: 'rgba(102, 102, 102, 0.5)'
                },
                markers: {
                    color: 'rgba(102, 102, 102, 0.5)'
                }
            },
            labels: {
                color: 'rgb(101, 101, 101)',
                font: '14px serif'
            }
        },
        seriesDefaults: {
            boxPlot: {
                downColor: 'rgba(0, 0, 0, 0.08)',
                mean: {
                    color: 'rgb(246, 246, 246)'
                },
                median: {
                    color: 'rgb(246, 246, 246)'
                },
                whiskers: {
                    color: 'rgb(255, 99, 88)'
                }
            },
            bullet: {
                target: {
                    color: 'rgb(101, 101, 101)'
                }
            },
            candlestick: {
                downColor: 'rgb(101, 101, 101)',
                line: {
                    color: 'rgb(101, 101, 101)'
                }
            },
            errorBars: {
                color: 'rgba(0, 0, 0, 0.5)'
            },
            horizontalWaterfall: {
                line: {
                    color: 'rgba(0, 0, 0, 0.08)'
                }
            },
            icon: {
                border: {
                    color: 'rgba(0, 0, 0, 0.08)'
                }
            },
            labels: {
                background: 'rgb(255, 255, 255)',
                color: 'rgb(101, 101, 101)',
                opacity: 0.8,
                font: '12px serif'
            },
            notes: {
                icon: {
                    background: 'rgba(0, 0, 0, 0.5)',
                    border: {
                        color: 'rgba(0, 0, 0, 0.5)'
                    }
                },
                line: {
                    color: 'rgba(0, 0, 0, 0.5)'
                },
                label: {
                    font: '14px serif'
                }
            },
            verticalBoxPlot: {
                downColor: 'rgba(0, 0, 0, 0.08)',
                mean: {
                    color: 'rgb(246, 246, 246)'
                },
                median: {
                    color: 'rgb(246, 246, 246)'
                },
                whiskers: {
                    color: 'rgb(255, 99, 88)'
                }
            },
            verticalBullet: {
                target: {
                    color: 'rgb(101, 101, 101)'
                }
            },
            waterfall: {
                line: {
                    color: 'rgba(0, 0, 0, 0.08)'
                }
            },
            area: {
                opacity: 0.8
            }
        },
        title: {
            color: 'rgb(101, 101, 101)',
            font: '16px serif'
        },
        seriesColors: [
            'rgb(255, 99, 88)',
            'rgb(255, 210, 70)',
            'rgb(120, 210, 55)',
            'rgb(40, 180, 200)',
            'rgb(45, 115, 245)',
            'rgb(170, 70, 190)'
        ]
    });
};
