// @progress/kendo-angular-charts/dist/npm/common/settings.component.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var configuration_service_1 = require("./configuration.service");
var copy_changes_1 = require("./copy-changes");
/**
 * @hidden
 */
var SettingsComponent = /** @class */ (function () {
    function SettingsComponent(configKey, configurationService) {
        this.configKey = configKey;
        this.configurationService = configurationService;
        this.store = {};
        if (configKey === undefined) {
            throw new Error('Configuration key not set');
        }
    }
    SettingsComponent.prototype.ngOnDestroy = function () {
        this.store = undefined;
        this.notify();
    };
    SettingsComponent.prototype.ngOnChanges = function (changes) {
        copy_changes_1.copyChanges(changes, this.store);
        this.notify();
    };
    SettingsComponent.prototype.markAsVisible = function () {
        this.store.visible = true;
        this.notify();
    };
    SettingsComponent.prototype.notify = function () {
        this.configurationService.notify(new configuration_service_1.Change(this.configKey, this.store));
    };
    return SettingsComponent;
}());
exports.SettingsComponent = SettingsComponent;
