// @progress/kendo-angular-charts/dist/es/events/instance-event.service.js
import { AxisLabelClickEvent } from './axis-label-click-event';
import { DragEvent } from './drag-event';
import { DragEndEvent } from './drag-end-event';
import { DragStartEvent } from './drag-start-event';
import { LegendItemHoverEvent } from './legend-item-hover-event';
import { LegendItemLeaveEvent } from './legend-item-leave-event';
import { NoteClickEvent } from './note-click-event';
import { NoteHoverEvent } from './note-hover-event';
import { NoteLeaveEvent } from './note-leave-event';
import { PaneRenderEvent } from './pane-render-event';
import { PlotAreaClickEvent } from './plot-area-click-event';
import { PlotAreaHoverEvent } from './plot-area-hover-event';
import { PlotAreaLeaveEvent } from './plot-area-leave-event';
import { RenderEvent } from './render-event';
import { SelectEvent } from './select-event';
import { SelectEndEvent } from './select-end-event';
import { SelectStartEvent } from './select-start-event';
import { SeriesClickEvent } from './series-click-event';
import { SeriesHoverEvent } from './series-hover-event';
import { SeriesOverEvent } from './series-over-event';
import { SeriesLeaveEvent } from './series-leave-event';
import { ZoomEvent } from './zoom-event';
import { ZoomEndEvent } from './zoom-end-event';
import { ZoomStartEvent } from './zoom-start-event';
var EVENT_MAP = {
    axisLabelClick: AxisLabelClickEvent,
    drag: DragEvent,
    dragEnd: DragEndEvent,
    dragStart: DragStartEvent,
    legendItemHover: LegendItemHoverEvent,
    legendItemLeave: LegendItemLeaveEvent,
    noteClick: NoteClickEvent,
    noteHover: NoteHoverEvent,
    noteLeave: NoteLeaveEvent,
    paneRender: PaneRenderEvent,
    plotAreaClick: PlotAreaClickEvent,
    plotAreaHover: PlotAreaHoverEvent,
    plotAreaLeave: PlotAreaLeaveEvent,
    render: RenderEvent,
    select: SelectEvent,
    selectEnd: SelectEndEvent,
    selectStart: SelectStartEvent,
    seriesClick: SeriesClickEvent,
    seriesHover: SeriesHoverEvent,
    seriesOver: SeriesOverEvent,
    seriesLeave: SeriesLeaveEvent,
    zoom: ZoomEvent,
    zoomEnd: ZoomEndEvent,
    zoomStart: ZoomStartEvent
};
/**
 * @hidden
 */
var InstanceEventService = /** @class */ (function () {
    function InstanceEventService() {
    }
    InstanceEventService.prototype.create = function (name, args, sender) {
        if (EVENT_MAP[name]) {
            return new EVENT_MAP[name](args, sender);
        }
    };
    return InstanceEventService;
}());
export { InstanceEventService };
