// @progress/kendo-angular-charts/dist/npm/stock-chart/navigator/category-axis.component.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var core_1 = require("@angular/core");
var configuration_service_1 = require("../../common/configuration.service");
var prefix_configuration_service_1 = require("../../common/prefix-configuration.service");
var category_axis_item_component_1 = require("../../chart/category-axis-item.component");
/**
 * The configuration component of the navigator category axis.
 *
 * @example
 *
 * ```html-no-run
 * <kendo-stockchart>
 *   <kendo-chart-navigator>
 *     <kendo-chart-navigator-category-axis
 *       color="maroon"
 *       [labels]="{color: 'green'}">
 *     </kendo-chart-navigator-category-axis>
 *   </kendo-chart-navigator>
 * </kendo-stockchart>
 * ```
 */
var NavigatorCategoryAxisComponent = /** @class */ (function (_super) {
    tslib_1.__extends(NavigatorCategoryAxisComponent, _super);
    function NavigatorCategoryAxisComponent(configurationService) {
        var _this = _super.call(this, configurationService, null) || this;
        _this.configurationService = configurationService;
        return _this;
    }
    NavigatorCategoryAxisComponent.decorators = [
        { type: core_1.Component, args: [{
                    changeDetection: core_1.ChangeDetectionStrategy.OnPush,
                    providers: [{
                            provide: prefix_configuration_service_1.PREFIX,
                            useValue: 'navigator.categoryAxis'
                        }, {
                            provide: configuration_service_1.ConfigurationService,
                            useClass: prefix_configuration_service_1.PrefixConfigurationService
                        }],
                    selector: 'kendo-chart-navigator-category-axis',
                    template: ''
                },] },
    ];
    /** @nocollapse */
    NavigatorCategoryAxisComponent.ctorParameters = function () { return [
        { type: configuration_service_1.ConfigurationService, },
    ]; };
    return NavigatorCategoryAxisComponent;
}(category_axis_item_component_1.CategoryAxisItemComponent));
exports.NavigatorCategoryAxisComponent = NavigatorCategoryAxisComponent;
